src/System.Windows.Extensions/src/System/Drawing/ImageConverter.cs

```diff
@@ -63,5 +75,75 @@ public override PropertyDescriptorCollection GetProperties(ITypeDescriptorContex
         }
 
         public override bool GetPropertiesSupported(ITypeDescriptorContext context) => true;
+
+        private unsafe Stream GetBitmapStream(byte[] rawData)
+        {
+            Debug.Assert(rawData != null, "rawData is null.");
+            try
+            {
+                fixed (byte* pByte = rawData)
+                {
+                    IntPtr addr = (IntPtr)pByte;
+
+                    if (addr == IntPtr.Zero)
+                    {
+                        return null;
+                    }
+
+                    // This will be pHeader.signature, but we try to
+                    // do this first so we avoid building the structure when we shouldn't
+                    if (rawData.Length <= sizeof(OBJECTHEADER) || Marshal.ReadInt16(addr) != 0x1c15)
+                    {
+                        return null;
+                    }
+
+                    // The data is in the form of OBJECTHEADER. It's an encoded format that Access uses to push imagesinto the DB.
+                    OBJECTHEADER pHeader = (OBJECTHEADER)Marshal.PtrToStructure(addr, typeof(OBJECTHEADER));
+
+                    // "PBrush" should be the 6 chars after position 12 as well.
+                    if (rawData.Length <= pHeader.headersize + 18 || Encoding.ASCII.GetString(rawData, pHeader.headersize + 12, 6) != "PBrush")
+                    {
+                        return null;
+                    }
+
+                    // We can safely trust that we've got a bitmap.
+                    byte[] searchArray = Encoding.ASCII.GetBytes("BM");
+
+                    // search for "BM" in the data which is the start of our bitmap data.
+                    // 18 is from (12+6) above.
+                    for (int i = pHeader.headersize + 18; i < pHeader.headersize + 510 && i + 1 < rawData.Length; i++)
+                    {
+                        if (searchArray[0] == pByte[i] &&
+                            searchArray[1] == pByte[i + 1])
+                        {
+                            return new MemoryStream(rawData, i, rawData.Length - i);
+                        }
+                    }
+                }
+            }
+            catch (OutOfMemoryException) // This exception may be caused by creating a new MemoryStream.
+            {
+            }
+            catch (ArgumentException) // This exception may be caused by Marshal.PtrToStructure.
+            {
+            }
+
+            return null;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct OBJECTHEADER
+        {
+            public short signature; // it's always 0x1c15
+            public short headersize;
+            public short objectType;
+            public short nameLen;
+            public short classLen;
+            public short nameOffset;
+            public short classOffset;
+            public short width;
+            public short height;
+            public IntPtr pInfo;
```

Comment:
You _should_ be able to calculate the precise offset of "BM" by utilizing nameLen & classLen. Would avoid scanning for it.

Reply:
how is "BM" related namelen and classlen ?is there any format of how the bytes are structured after the header ?

Reply:
"BM" is the start of a normal bitmap file. So we're looking for the normal file start after the OLE header.

Reply:
@JeremyKuhne I think its always 78 for bitmap files https://stackoverflow.com/questions/2874403/strip-ole-header-information-ms-access-sql-server

Reply:
I left a more detailed comment on the issue related to the structure.